modules/account-lib/test/unit/coin/hbar/transaction.ts

```diff
@@ -0,0 +1,74 @@
+import should from 'should';
+import { coins } from '@bitgo/statics';
+import { Transaction } from '../../../../src/coin/hbar/transaction';
+import * as testData from '../../../resources/hbar/hbar';
+import { KeyPair } from '../../../../src/coin/hbar/keyPair';
+
+describe('Hbar Transaction', () => {
+  const coin = coins.get('thbar');
+
+  /**
+   *
+   */
+  function getTransaction(): Transaction {
+    return new Transaction(coin);
+  }
+
+  it('should throw empty transaction', () => {
+    const tx = getTransaction();
+    should.throws(() => {
+      tx.toJson();
+    });
+    should.throws(() => {
+      tx.toBroadcastFormat();
+    });
+  });
+
+  describe('should sign if transaction is', () => {
+    it('invalid', function() {
+      const tx = getTransaction();
+      return tx.sign(testData.INVALID_KEYPAIR_PRV).should.be.rejected();
+    });
+
+    it('valid', async () => {
+      const tx = getTransaction();
+      tx.bodyBytes(testData.WALLET_TXDATA);
+      const keypair = new KeyPair({ prv: testData.ACCOUNT_1.privateKey });
+      await tx.sign(keypair).should.be.fulfilled();
+      should.equal(
+        new Buffer(tx.hederaTx.sigMap!.sigPair![0].pubKeyPrefix!).toString('hex'),
+        testData.ACCOUNT_1.publicKey.slice(24),
+      );
+    });
+
+    it('multiple valid', async () => {
+      const tx = getTransaction();
+      tx.bodyBytes(testData.WALLET_TXDATA);
+      const keypair = new KeyPair({ prv: testData.ACCOUNT_1.privateKey });
+      const keypair2 = new KeyPair({ prv: testData.OPERATOR.privateKey });
+      await tx.sign(keypair).should.be.fulfilled();
+      should.equal(
+        new Buffer(tx.hederaTx.sigMap!.sigPair![0].pubKeyPrefix!).toString('hex'),
+        testData.ACCOUNT_1.publicKey.slice(24),
+      );
+      await tx.sign(keypair2).should.be.fulfilled();
+      should.equal(
+        new Buffer(tx.hederaTx.sigMap!.sigPair![0].pubKeyPrefix!).toString('hex'),
+        testData.ACCOUNT_1.publicKey.slice(24),
+      );
+      should.equal(
+        new Buffer(tx.hederaTx.sigMap!.sigPair![1].pubKeyPrefix!).toString('hex'),
+        testData.OPERATOR.publicKey.slice(24),
+      );
+    });
+  });
+
+  describe('should return encoded tx', function() {
+    it('valid sign', async function() {
+      const tx = getTransaction();
+      tx.bodyBytes(testData.WALLET_TXDATA);
+      await tx.sign(testData.KEYPAIR_PRV);

```

Comment:
We usually consume the `sign` method from the `txBuilder`, not the tx itself. We should test that too if we haven't.
We may need something like `addSignature` in case we produce the signature offline, i.e.: https://github.com/BitGo/BitGoJS/blob/0d800b724b32bf15346a8f2adc2a48ae43ff3c57/modules/account-lib/src/coin/xtz/transaction.ts#L223

Reply:
We have a few tests for signature using the txBuilder in another PR (#834) 
I think it's better to land those tests in that PR to avoid conflicts since both PR are ready.

Reply:
I think this is resolved now